lib/std/array_list.zig

```diff
@@ -90,16 +106,18 @@ pub fn ArrayListAligned(comptime T: type, comptime alignment: ?u29) type {
         /// ArrayList takes ownership of the passed in slice. The slice must have been
         /// allocated with `allocator`.
         /// Deinitialize with `deinit` or use `toOwnedSlice`.
+        /// **Causes illegal behavior when `slice.len` is `std.math.maxInt(usize)`.**
         pub fn fromOwnedSliceSentinel(allocator: Allocator, comptime sentinel: T, slice: [:sentinel]T) Self {
             return Self{
                 .items = slice,
-                .capacity = slice.len + 1,
+                .capacity = slice.len + 1, // Overflow: API misuse?
```

Comment:
Comments should not ask questions like this.

Reply:
What's the status quo on this? Are sentineled slices where `len` is `std.math.maxInt(usize)` valid?

Reply:
sentineled slices guarantee a sentinel at len + 1, therefore slice.len + 1 for sentineled slices is impossible to overflow.